app/src/main/java/com/serwylo/lexica/view/LexicaView.java

```diff
@@ -195,24 +195,66 @@ private void drawTimer(Canvas canvas) {
 	private int drawWordCount(Canvas canvas, int left, int top, int bottom) {
 		p.setTypeface(Typeface.SANS_SERIF);
 		p.setARGB(255, 0, 0, 0);
+		float actualBottom = top;
 
-		float textSize = (bottom - top - paddingSize) / 4f;
-		if (textSize > textSizeNormal) {
-			textSize = textSizeNormal;
-		}
+		if (!game.showBreakdown()) {
+			float textSize = (bottom - top - paddingSize) / 4f;
+			if (textSize > textSizeNormal) {
+				textSize = textSizeNormal;
+			}
+			p.setTextSize(textSize);
 
-		p.setTextSize(textSize);
-		float actualBottom = top + textSize;
-		canvas.drawText(getContext().getString(R.string.score), left, actualBottom, p);
+			actualBottom += textSize;
+			canvas.drawText(getContext().getString(R.string.score), left, actualBottom, p);
 
-		actualBottom += textSize;
-		canvas.drawText(Integer.toString(game.getScore()), left, actualBottom, p);
+			actualBottom += textSize;
+			canvas.drawText(Integer.toString(game.getScore()), left, actualBottom, p);
 
-		actualBottom += paddingSize + textSize;
-		canvas.drawText(game.getWordCount() + "/" + game.getMaxWordCount(), left, actualBottom, p);
+			actualBottom += paddingSize + textSize;
+			canvas.drawText(game.getWordCount() + "/" + game.getMaxWordCount(), left, actualBottom, p);
+
+			actualBottom += textSize;
+			canvas.drawText(getContext().getString(R.string.words), left, actualBottom, p);
+		} else {
+			int[] maxWordCounts = game.getMaxWordCountsByLength();
+			int lines = 0;
+			int lenPad = 1;
+			int countPad = 2;
+			for (int length = 3; length < maxWordCounts.length; length++) {
+				int count = maxWordCounts[length];
+				if (count > 0) {
+					lines++;
+					if (count > 99) {
+						countPad = 3;
+					}
+					if (length > 9) {
+						lenPad = 2;
+					}
+				}
+			}
+			float textSize = (bottom - top) / lines;
+			if (textSize > textSizeNormal) {
+				textSize = textSizeNormal;
+			}
+			if (textSize < getResources().getDimension(R.dimen.textSizeMinimum)) {
+				textSize = getResources().getDimension(R.dimen.textSizeMinimum);
+			}
+			p.setTextSize(textSize);
+			p.setTypeface(Typeface.MONOSPACE);
+
+			int[] wordCounts = game.getWordCountsByLength();
+			for (int length = 3; length < maxWordCounts.length; length++) {
+				int count = maxWordCounts[length];
+				if (count > 0) {
+					actualBottom += textSize;
+					String scoreLine = pad(length, lenPad) + ":"
```

Comment:
I'm not sure how this concatenation plays for people with RTL languages. However there are other places which suffer from this as well in Lexica, so that is probably a story for another day.